sdks/python/apache_beam/metrics/metric_test.py

```diff
@@ -162,7 +161,7 @@ def process(self, element):
     # Verify user counter.
     metric_results = (
         res.metrics().query(
-            MetricsFilter().with_name('metrics_user_counter_element')))
+            MetricsFilter().with_metric(SomeDoFn.user_counter_elements)))
```

Comment:
Should we add a new test for with_metric instead of replacing the with_name test?

Reply:
yes, can you add both to the test, please?

Reply:
Sure. I pulled the old test back.